syntax = "proto2";

package SulDownloaderProto;


message ProductStatusReport
{
    optional string rigidName = 1;
    optional string productName = 2;
    optional string installedVersion = 3;
    optional string downloadVersion = 4;
    optional string errorDescription = 7;

    enum ProductStatus
    {
       SYNCFAILED = 0;
       INSTALLFAILED = 1;
       UNINSTALLFAILED = 2;
       VERIFYFAILED = 3;
       UPTODATE = 4;
       UPGRADED = 5;
       UNINSTALLED = 6;
    }
    optional ProductStatus productStatus = 9 [default=UPTODATE];
}

message WarehouseComponent
{
    optional string rigidName = 1;
    optional string productName = 2;
    optional string installedVersion = 3;
    optional string warehouseVersion = 4;
}

message DownloadStatusReport
{
   optional string startTime = 1;
   optional string finishTime = 2;
   optional string status =4;
   optional string sulError = 5;
   optional string errorDescription = 6;
   optional string urlSource = 7;
   repeated ProductStatusReport products = 8;
   repeated WarehouseComponent warehouseComponents = 9;
   optional bool supplementOnlyUpdate = 10 [default=false];
}
